publications.php

```diff
@@ -1,59 +1,83 @@
-$researchers=new NGIresearchers();
-$publications=new NGIpublications();
 
 if($USER->auth>0) {
-	if(isset($_GET['id'])) {
-		$output=$publications->showPublication($_GET['id']);
-	} else {
-		if(filter_var($_REQUEST['year'],FILTER_VALIDATE_INT)) {
-			$output=$publications->showPublicationList($_REQUEST['year']);
-		} else {
-			$output=$publications->showPublicationList(date("Y"));
-		}
-
-		//----------------------------------------------------
-		// "Sync" with SciLifeLab Publications database
-
-		/*
-		// JSON sources for Production and Applications papers in the SciLifeLab database
-		$sources=array(
-			'https://publications.scilifelab.se/label/NGI%20Stockholm%20%28Genomics%20Production%29.json', 
-			'https://publications.scilifelab.se/label/NGI%20Stockholm%20%28Genomics%20Applications%29.json'
-		);
-		$list=$publications->checkDB($sources);
-		
-		// Auto add missing papers
-		$autoadd_data=$pubmed->summary($list['missing']);
-		
-		// OBS! This will NOT parse authors since no lab is given, however this will be done if the PMID shows up in searches
-		// Not a big issue since this is likely only a one-time event and only for old papers
-		$add_papers=$publications->addBatch($autoadd_data,FALSE); 
-
-		//These two papers are listed in publications db but classified here as "discarded" - double check these!
-		// 27913302: NGI labels removed in pub db (seq on HiSeq 1500)
-
-		if(count($list['mismatch'])>0) {
-			$missing=implode(',', $list['mismatch']);
-			$query=sql_query("SELECT * FROM publications WHERE pmid IN($missing)");
-			$output=$publications->showSelectedPublications($query);
-		}
-		*/
-		
-		//----------------------------------------------------
-		// Examples
-		
-		//$publist=$publications->listPublications(2017);
-		//$output=$publications->formatPubList($publist);
-
-		//$query=sql_query("SELECT * FROM publications WHERE status IS NULL AND score>5 AND pubdate>='2017-01-01' ORDER BY score DESC");
-		//$query=sql_query("SELECT * FROM publications WHERE status='maybe' AND pubdate>='2017-01-01' ORDER BY score DESC");
-		//$output=$publications->showSelectedPublications($query);
-
-		//----------------------------------------------------
+	$publications=new NGIpublications();
+
+	$years=range(2010, date('Y')+1);
+	$years_select=array_combine($years,$years);
+	$years_select[0]="All";
+	asort($years_select);
+
+	$filterform=new htmlForm("publications.php","get",4);
+	$filterform->addSelect("Status","status",array('all' => 'All', 'verified' => 'Verified', 'discarded' => 'Discarded', 'maybe' => 'Maybe', 'auto' => 'Auto'),$_GET);
+	$filterform->addSelect("Year","year",$years_select,$_GET);
+	$filterform->addSelect("Order by","order_by",array('score' => 'Score', 'pubdate' => 'Publication date'),$_GET);
+	$filterform->addSelect("Sort","sort",array('desc' => 'Descending', 'asc' => 'Ascending'),$_GET);
+	$filterform->addInput("",array('type' => 'submit', 'name' => 'submit', 'value' => 'Filter search', 'class' => 'button'));
+
```

Comment:
I think we talked about searching for text also? Author names and maybe titles? I.e. did group X publish those cool results or not?

But that is something for a later date.

Reply:
Yes that would be good to add, I think I'll start adding issues so I don't forget stuff